include/openthread/platform/dnssd.h

```diff
@@ -0,0 +1,322 @@
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file includes the platform abstraction for the DNS-SD (e.g., mDNS) on infrastructure network.
+ *
+ */
+
+#ifndef OPENTHREAD_PLATFORM_DNSSD_H_
+#define OPENTHREAD_PLATFORM_DNSSD_H_
+
+#include <stdint.h>
+
+#include <openthread/dns.h>
+#include <openthread/error.h>
+#include <openthread/instance.h>
+#include <openthread/ip6.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup plat-dns-sd
+ *
+ * @brief
+ *   This module includes the platform abstraction for the DNS-SD (e.g., mDNS) on infrastructure network.
+ *
+ * @{
+ *
+ * The DNS-SD platform API are used only when `OPENTHREAD_CONFIG_PLATFORM_DNSSD_ENABLE` is enabled.
+ *
+ */
+
+/**
+ * Represents state of DNS-SD platform.
+ *
+ */
+typedef enum otPlatDnssdState
+{
+    OT_PLAT_DNSSD_STOPPED, ///< Stopped and unable to register any service or host.
+    OT_PLAT_DNSSD_READY,   ///< Running and ready to register service or host.
+} otPlatDnssdState;
+
+/**
+ * Represents a request ID (for registering/unregistering a service or host).
+ *
+ */
+typedef uint32_t otPlatDnssdRequestId;
+
+/**
+ * Represents callback function used when registering/unregistering a host or service.
+ *
+ * See `otPlatDnssdRegisterService()`, `otPlatDnssdUnregisterService()`, `otPlatDnssdRegisterHost()`, and
+ * `otPlatDnssdUnregisterHost()` for more details about when to invoke the callback and the `aError` values that can
+ * be returned in each case.
+ *
+ * @param[in] aInstance     The OpenThread instance.
+ * @param[in] aRequestId    The request ID.
+ * @param[in] aError        Error indicating the outcome of request.
+ *
+ */
+typedef void (*otPlatDnssdRegisterCallback)(otInstance *aInstance, otPlatDnssdRequestId aRequestId, otError aError);
+
+/**
+ * Represent a DNS-SD service.
+ *
+ * See `otPlatDnssdRegisterService()`, `otPlatDnssdUnregisterService()` for more details about fields in each case.
+ *
+ */
+typedef struct otPlatDnssdService
+{
+    const char        *mHostName;            ///< The host name (does not include domain name).
+    const char        *mServiceInstance;     ///< The service instance name label (not the full name).
+    const char        *mServiceName;         ///< The service name (e.g., "_mt._udp", does not include domain name).
```

Comment:
I think `mServiceType` should be more accurate. "Service Name" standards for only "_mt" in the example

Reply:
This also aligns with the name "SubType"

Reply:
Good suggestion. Thanks. I renamed it to `mServiceType` in the new push.